packages/ui/src/Popup/Accounts/NetworkSelector/NetworkEdit.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { FC, ChangeEvent, useState } from 'react';
+import { toast } from 'react-toastify';
+import { Text, Flex, Box, Icon, TextInput } from "@polymeshassociation/extension-ui/ui"
+import { SvgChecklistFilled, SvgAlertCircle } from '@polymeshassociation/extension-ui/assets/images/icons';
+import { NetworkLabelContainer, NetworkLabel } from './styles';
+
+interface Props {
+  defaultValue: string;
+  setUrlValue: (customNetworkUrl: string) => void;
+}
+
+const rpcUrlRegex =
+  /(http|ws)s?:\/\/(www\.)?[-a-zA-Z0-9@:%._+~#=]{1,256}\.[a-zA-Z0-9()]{2,63}\b([-a-zA-Z0-9()@:%_+.~#?&=]*)/;
+
+
+export const NetworkEdit: FC<Props> = ({
+  defaultValue,
+  setUrlValue,
+}) => {
+  const [value, setValue] = useState(defaultValue);
+
+  const handleChangeValue = (e: ChangeEvent<HTMLInputElement>) => {
+    setValue(e.target.value.trim());
+  };
+
+  const handleSetValue = () => {
+    const networkUrl = value[value.length - 1] === '/' ? value : `${value}/`
```

Comment:
I've added this check because the url from Adam `wss://dev-fsn001.nsite.dev/staging/` doesn't work without this slash.
So i guess we need to update defaults for consistency